### PR TITLE
Follow-ups about IContainerRuntimeOptionsInternal and flushMode option

### PR DESCRIPTION
**Pull request by markfields**
## Description

These are follow-up changes to #23272 

* Some post-merge PR discussions
* Removing usages of `flushMode` so it can be removed when the time comes

### CODE REVIEW

File: packages/runtime/container-runtime/src/test/containerRuntime.spec.ts [hunk 1]
```diff
@@ -1490,6 +1490,10 @@ describe("Runtime", () => {
 				};
 			};
 
+			const runtimeOptions: IContainerRuntimeOptionsInternal = {
+				flushMode: FlushModeExperimental.Async as unknown as FlushMode,
```
Comment (jason-ha) [inline packages/runtime/container-runtime/src/test/containerRuntime.spec.ts:1494]: Looks like another opportunity. IContainerRuntimeOptionsInternal should just allow FlushModeExperimental.Async, yeah?
But I am hoping(?) that TypeScript doesn't allow that now.?

**Review by jason-ha — APPROVED** [review]

File: packages/runtime/container-runtime/src/test/containerRuntime.spec.ts (hunk 1 above)
Comment (markfields) [reply packages/runtime/container-runtime/src/test/containerRuntime.spec.ts:1494]: Right, would be a future improvement to bring that one into the internal flushMode option. Now that it's all internal we can do that at our leisure. Not sure who is using/working on that Async mode.

**Comment by markfields** [comment]
@jatgarg, @anthony-murphy  -- can you approve for CR-API group?

**Review by Josmithr — APPROVED** [review]
API changes look good

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 2]
```diff
@@ -1517,7 +1517,10 @@ export class ContainerRuntime
 		electedSummarizerData: ISerializedElection | undefined,
 		chunks: [string, string[]][],
 		dataStoreAliasMap: [string, string][],
-		runtimeOptions: Readonly<Required<IContainerRuntimeOptions>>,
+		runtimeOptions: Readonly<
+			Required<Omit<IContainerRuntimeOptions, "flushMode" | "enableGroupedBatching">> &
```
Comment (markfields) [inline packages/runtime/container-runtime/src/containerRuntime.ts:1521]: @jason-ha, @anthony-murphy, @Josmithr -- I'm moving forward with merging this, but to be pedantic, I don't think this kind of change is in general perfectly backwards compatible. Wanted to call it out to check my work and for transparency.

If someone is using `typeof ContainerRuntime` to type a class declaration, their implementation may theoretically depend on `flushMode` being `Required`, but then after this change a caller could stop passing it.  Here's a [TS Playground example](https://www.typescriptlang.org/play/?#code/MYewdgzgLgBANiA5jAvDUkRwKYDoGIDcAUMcHAIYQQwBCV2MA3sTGzAPQcwDyAMgBEAXOnDQATgFdgUEOIAUAMxAgRTCiLCSAtgCNs4gDQxdmnfvEBfAJTNLrdlxgA5AKIB1IQ7YYJ02QrKqswaMFp6Bsa6APxmEVa2TPb2ZGKwPFAAFgYAwpTUIlAAngAO2CCKdAyo6Pk0LOyc3ACqENg0WYxYACYwEACWiGAUUJLi2MaSwwDuFOMwWSMw-bCZVOhrYIjY3d6ikFBSMnJKKmqh4RZRcRY2zHuNvrDqUZY1QSSNjQTyFFHWn3YyRIxCcOQocDg-S2om6jH6NDWJRKRQWIBgIG0KxMJgYvXAC2yYWw0wWpWhyAqVTaxDAJN4nXEeSoEHkTBgoQAjDAbCQgA).

**Pull request merged**